framework/contracts/account/account/src/contract.rs

```diff
@@ -54,19 +56,25 @@ pub fn instantiate(
     // Use CW2 to set the contract version, this is needed for migrations
     cw2::set_contract_version(deps.storage, ACCOUNT, CONTRACT_VERSION)?;
 
-    let account_id =
-        account_id.unwrap_or_else(|| /*  TODO: Query VC for Sequence*/ AccountId::local(0));
+    // ## Manage ##
+    let module_factory_address = deps.api.addr_validate(&module_factory_address)?;
+    let version_control_address = deps.api.addr_validate(&version_control_address)?;
+
+    let account_id = account_id.unwrap_or_else(|| {
+        AccountId::local(
+            LOCAL_ACCOUNT_SEQUENCE
+                .query(&deps.querier, version_control_address.clone())
+                // default to 0 for first account
+                .unwrap_or_default(),
```

Comment:
Isn't it dangerous that any error will turn the account sequence to 0 on instantiation ? I think we should be more careful on the error handling here !

Reply:
This could mean parsing the error or simply initializing the account sequence on the version control on instantiation

Reply:
Think that second option makes the most sense :) good comment

Reply:
Done